tests/stm32/Cargo.toml

```diff
@@ -14,6 +14,7 @@ stm32f429zi = ["embassy-stm32/stm32f429zi", "chrono", "eth", "stop", "can", "not
 stm32f446re = ["embassy-stm32/stm32f446re", "chrono", "stop", "can", "not-gpdma", "dac", "sdmmc"]
 stm32f767zi = ["embassy-stm32/stm32f767zi", "chrono", "not-gpdma", "eth", "rng"]
 stm32g071rb = ["embassy-stm32/stm32g071rb", "cm0", "not-gpdma", "dac"]
+stm32g431cb = ["embassy-stm32/stm32g431cb", "chrono", "stop", "not-gpdma", "rng", "fdcan"]
```

Comment:
yeah this will do nothing because the HIL test farm doesn't have a stm32g431cb. Please add it to the existing chips instead. If there's any wire I have to add for loopback please let me know.

Reply:
Loopback can be configured internally on the part, it will still output signals on the TX pin so this just needs to be set to use a TX pin that is unused or won't interfere with anything else that is connected.

Reply:
it's better to use external loopback if possible, because it also tests the GPIO AF configuration etc are working correctly. it's "closer" to normal mode.

Reply:
With CAN you would need to have transceivers and have two separate instances talk to each other to test a full loopback externally otherwise there is not another node on the bus to ACK the message and it will consider the transmission failed. With internal loopback it ignores the missing ACK.

Reply:
I believe this could be improved with a CAN frame filtering. Configure a filter or two, send a message - match a filter - and receive it. Then send another NOT matching message and assert that it's not being received. This should cover a lot of CAN API without touching hardware.

Reply:
On a very short distance it's possible to do a CAN without transceivers AFAIK - using a diode instead. For testing it should be possible.